packages/zosfiles/src/methods/upload/Upload.ts

```diff
@@ -487,6 +488,12 @@ export class Upload {
             apiResponse.etag = uploadRequest.response.headers.etag;
         }
 
+        if (options.encoding != null) {
+            await Utilities.chtag(session, origUssname, Tag.TEXT, options.encoding);
+        } else if (options.binary) {
+            await Utilities.chtag(session, origUssname, Tag.BINARY);
```

Comment:
I'm afraid that the file on the mainframe isn't tagged after testing.


```
- untagged    T=off -rw-r--r--    1 FERNANDO 0           4163 Dec  6 15:30 test-file1.txt
```
I could be doing something wrong though.